packages/hoprd/crates/hoprd-hoprd/src/main.rs

```diff
@@ -0,0 +1,176 @@
+pub mod api;
+pub mod cli;
+pub mod config;
+pub mod errors;
+pub mod token;
+
+use std::{sync::Arc, time::SystemTime, path::Path};
+
+use async_lock::RwLock;
+use chrono::{DateTime, Utc};
+
+use core_transport::HalfKeyChallenge;
+use hoprd_keypair::key_pair::{IdentityOptions, HoprKeys};
+use utils_log::{info, warn};
+use utils_types::traits::{PeerIdLike, ToHex};
+use crate::cli::CliArgs;
+
+
+const ONBOARDING_INFORMATION_INTERVAL: std::time::Duration = std::time::Duration::from_secs(30);
+
+    // export function removeAllInPath(target: string) {
+    //   const p = path.normalize(target)
+    //   rmSync(p, { recursive: true, force: true })
+    // }
+
+    // function stopGracefully(signal) {
+    //   log(`Process exiting with signal ${signal}`)
+    //   process.exit()
+    // }
+
+    // process.on('uncaughtExceptionMonitor', (err, origin) => {
+    //   // Make sure we get a log.
+    //   log(`FATAL ERROR, exiting with uncaught exception`, origin, err)
+    // })
+
+    // process.once('exit', stopGracefully)
+    // process.on('SIGINT', stopGracefully)
+    // process.on('SIGTERM', stopGracefully)
+
+    // // Metrics
+    // const metric_latency = create_histogram_with_buckets(
+    //   'hoprd_histogram_message_latency_ms',
+    //   'Histogram of measured received message latencies',
+    //   new Float64Array([10.0, 25.0, 50.0, 100.0, 250.0, 500.0, 1000.0, 2500.0, 5000.0, 10000.0, 20000.0])
+    // )
+
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = <CliArgs as clap::Parser>::parse();
+
+    // TODO: remove the dry run
+    if ! args.dry_run {
+        info!("This is HOPRd {}", hopr_lib::constants::APP_VERSION);
+        
+        if std::env::var("DAPPNODE").map(|v| v.to_lowercase() == "true").unwrap_or(false) {
+            info!("The HOPRd node appears to run on DappNode");
+        }
+
+        let cfg = config::HoprdConfig::from_cli_args(args, false)?;
+        info!("Node configuration: {}", cfg.as_redacted_string()?);
+
+        // Find or create an identity
+        let identity_opts = IdentityOptions {
+            initialize: cfg.hopr.db.initialize,
+            id_path: cfg.identity.file.clone(),
+            password: cfg.identity.password.clone(),
+            use_weak_crypto: Some(cfg.test.use_weak_crypto),
+            private_key: cfg.identity.private_key.clone().and_then(|v| cli::parse_private_key(&v).ok()),
+        };
+
+        let hopr_keys = HoprKeys::init(identity_opts)?;
+
+        info!("This node has packet key '{}' and uses a blockchain address '{}'",
+            core_transport::Keypair::public(&hopr_keys.packet_key).to_peerid_str(),
+            core_transport::Keypair::public(&hopr_keys.chain_key).to_hex()
+        );
+
+        if core_transport::Keypair::public(&hopr_keys.packet_key).to_string().starts_with("0xff") {
```

Comment:
Add a TODO here to remove this check once the PR with the fix is merged in 3.0

Reply:
This one? https://github.com/hoprnet/hoprnet/pull/5665